sleap/gui/commands.py

```diff
@@ -956,25 +953,53 @@ class ExportAnalysisFile(AppCommand):
     def do_action(cls, context: CommandContext, params: dict):
         from sleap.io.format.sleap_analysis import SleapAnalysisAdaptor
 
-        SleapAnalysisAdaptor.write(params["output_path"], context.labels)
+        for output_path, video in params["analysis_videos"]:
+            SleapAnalysisAdaptor.write(output_path, context.labels, video=video)
 
     @staticmethod
     def ask(context: CommandContext, params: dict) -> bool:
+        labels = context.labels
+        if len(labels.labeled_frames) == 0:
+            return False
+
+        if params["all_videos"]:
+            all_videos = context.labels.videos
+        else:
+            all_videos = [context.state["video"] or context.labels.videos[0]]
+
+        # Check for labeled frames in each video
+        videos = []
+        for video in all_videos:
+            lfs = labels.get(video)
+            if len(lfs) != 0:
+                videos.append(video)
+        if len(videos) == 0:
+            return False
+
         default_name = context.state["filename"] or "labels"
-        p = PurePath(default_name)
-        default_name = str(p.with_name(f"{p.stem}.analysis.h5"))
+        fn = PurePath(default_name)
 
-        filename, selected_filter = FileDialog.save(
-            context.app,
-            caption="Export Analysis File...",
-            dir=default_name,
-            filter="SLEAP Analysis HDF5 (*.h5)",
-        )
+        output_paths = []
+        analysis_videos = []
+        for video in videos:
```

Comment:
Not a fan of doing this in a loop here. It will pop up a dialog for every single video in the project -- some of which could have 30+.

If there is more than one video, this command should instead show a `FileDialog.openDir` to prompt the user to pick a directory to save everything to. They lose the ability to customize the filename, but that's way better than clicking "Save" 30+ times.

Reply:
Agreed - this could be very annoying. I changed this to open a `FileDialog.openDir` when multiple videos exist, otherwise use `FileDialog.save` for single video projects.